mlir/utils/jenkins/miopen-tests/miopen_validate.sh

```diff
@@ -125,8 +136,12 @@ function setup_environment() {
         4) MIOPEN_DEBUG_FIND_ONLY_SOLVER=ConvMlirIgemmWrW ;;
         *) echo "$0: Unsupported direction flag $DIRECTION"; exit 2
     esac
-    if [[ $XDLOPS == 1 ]]; then
+    local xdlops
+    xdlops=$(/opt/rocm/bin/rocm_agent_enumerator |grep gfx908)
```

Comment:
Is this really the right approach? It feels like we'd want to have xdlops be something we can pass in instead of relying on autodetection.

That is, I'm curious about the logic here.

Reply:
The autodetection allows us to run xdlops tests on private CI and non-xdlops on public CI. That's what the performance test is doing. Thus we can test on more CI nodes and have a more balanced work on private and public CI.

Reply:
Good point, and this makes sense as a change.

Reply:
Though it might be nice to retain the ability to override `xdlops` for manual testing

Reply:
Good idea. I think I will have it in the next PR.

Reply:
Could you add it here?

The way I'd do this is to have the xdlops autodetection in options parsing - if the user doesn't pass in `--xdlops` or `--no-xdlops`, call the agent enumerator and detect a setting.